tolerance issue in test_BatchMaxGumbel

On this pipeline https://github.com/Rodrigo-Tenorio/distromax/pull/16/checks `test_BatchMaxGumbel()` failed with
```
E               AssertionError: 
E               Not equal to tolerance rtol=0.01, atol=0
E               
E               Mismatched elements: 1 / 1 (100%)
E               Max absolute difference: 0.01088712
E               Max relative difference: 0.01088712
E                x: array(1.010887)
E                y: array(1)
```

Seems this needs either
* higher tolerance
* fixed random seed to avoid sporadic bad cases
* flaky

/assign @Rodrigo-Tenorio

--- a/setup.py
+++ b/setup.py
@@ -27,7 +27,7 @@ setup(
         "examples": [
             "matplotlib",
         ],
-        "tests": ["pytest"],
+        "tests": ["pytest", "flaky"],
     },
     install_requires=[
         "numpy",

--- a/tests/test_distromax.py
+++ b/tests/test_distromax.py
@@ -11,6 +11,8 @@ import distromax
 
 import pytest
 
+from utils_for_testing import is_flaky
+
 HERE = Path(__file__).parent
 EXAMPLES = (HERE.parent / "examples").glob("*.py")
 
@@ -36,7 +38,7 @@ def test_AnalyticalGammaToGumbel():
         with pytest.raises(ValueError):
             distromax.analytical.AnalyticalGammaToGumbel(**input_values)
 
-
+@pytest.mark.flaky(max_runs=3, min_passes=1, rerun_filter=is_flaky)
 def test_BatchMaxGumbel(samples=None, fitting_class=None, fitting_class_kwargs=None):
 
     samples = samples if samples is not None else ground_truth_gumbel.rvs(size=(100000, 2))

--- a/tests/utils_for_testing.py
+++ b/tests/utils_for_testing.py
@@ -1,0 +1,5 @@
+class FlakyError(Exception):
+    pass
+
+def is_flaky(err, *args):
+    return issubclass(erro[0], FlakyError)

